### PR TITLE
More recursive schema tests for tree

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Test some of the fragile edge cases which are getting impacted by a proposed upcoming change.

This helps establish a baseline of what currently compiles.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts [hunk 1]
```diff
@@ -715,4 +756,130 @@ describe("SchemaFactory Recursive methods", () => {
 		const r = hydrate(Root, { r: new ArrayRecursive([]) });
 		assert.deepEqual([...r.r], []);
 	});
+
+	/**
+	 * These cases are anti-patterns which are documented not to work, but still might get used my someone.
```
Comment (jenn-le) [inline packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts:761]:
```suggestion
	 * These cases are anti-patterns which are documented not to work, but still might get used by someone.
```

File: packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts [hunk 2]
```diff
@@ -715,4 +756,130 @@ describe("SchemaFactory Recursive methods", () => {
...
+	 * SchemaFactory documents the non explicit sub-classing pattern (POJO mode) is not supported for recursive schema.
+	 * Additionally it documents this pattern is not supported for Explicitly named Maps and Arrays.
```
Comment (jenn-le) [inline packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts:763]:
```suggestion
	 * Additionally it documents this pattern is not supported for explicitly named Maps and Arrays.
```

File: packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts [hunk 3]
```diff
@@ -715,4 +756,130 @@ describe("SchemaFactory Recursive methods", () => {
...
+	 *
+	 * These tests violate both of these statements, seeing which version currently compile and which ones do not.
```
Comment (jenn-le) [inline packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts:765]:
```suggestion
	 * These tests violate both of these statements, seeing which versions currently compile and which ones do not.
```

File: packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts [hunk 4]
```diff
@@ -715,4 +756,130 @@ describe("SchemaFactory Recursive methods", () => {
...
+	 * Having these tests in place will help know when.if be further break cases like these to adi in writing helpful changesets for impacted customers.
```
Comment (jenn-le) [inline packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts:766]: The "when.if" part seems to be accidental.

File: packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts [hunk 5]
```diff
@@ -715,4 +756,130 @@ describe("SchemaFactory Recursive methods", () => {
...
+	 *
+	 * These patterns also [break type safety in .d.ts generation](https://github.com/microsoft/TypeScript/issues/55832):
+	 * this is one of the reasons they are not supported.
+	 * The import-testing has test coverage for aspect.
```
Comment (jenn-le) [inline packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts:770]: What does aspect mean here?

File: packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts [hunk 6]
```diff
@@ -715,4 +756,130 @@ describe("SchemaFactory Recursive methods", () => {
...
+	 * They also have poorer error quality and IntelliSense, and tend to fail to compile in some cases.
+	 *
+	 * These tests are all about the Typing.
```
Comment (jenn-le) [inline packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts:773]: Is this meant to be capitalized?

```suggestion
	 * These tests are all about the typing.
```

File: packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts [hunk 7]
```diff
@@ -715,4 +756,130 @@ describe("SchemaFactory Recursive methods", () => {
...
+	 * These cases are anti-patterns which are documented not to work, but still might get used by someone.
+	 * SchemaFactory documents the non explicit sub-classing pattern (POJO mode) is not supported for recursive schema.
+	 * Additionally it documents this pattern is not supported for explicitly named Maps and Arrays.
+	 *
+	 * These tests violate both of these statements, seeing which versions currently compile and which ones do not.
+	 * Having these tests in place will help know when.if be further break cases like these to adi in writing helpful changesets for impacted customers.
+	 *
+	 * These patterns also [break type safety in .d.ts generation](https://github.com/microsoft/TypeScript/issues/55832):
+	 * this is one of the reasons they are not supported.
+	 * The import-testing has test coverage for aspect.
```
Comment (CraigMacomber) [inline packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts:770]:
```suggestion
	 * The import-testing package has test coverage for aspect.
```

File: packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts (hunk 4 above)
Comment (CraigMacomber) [reply packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts:766]:
```suggestion
	 * Having these tests in place will help discover when/if further changes break cases like these to aid in writing helpful changesets for impacted customers using these unsupported patterns on accident.
```

File: packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts [hunk 8]
```diff
@@ -715,4 +756,130 @@ describe("SchemaFactory Recursive methods", () => {
...
+	 * These cases are anti-patterns which are documented not to work, but still might get used by someone.
+	 * SchemaFactory documents the non explicit sub-classing pattern (POJO mode) is not supported for recursive schema.
+	 * Additionally it documents this pattern is not supported for explicitly named Maps and Arrays.
+	 *
+	 * These tests violate both of these statements, seeing which versions currently compile and which ones do not.
+	 * Having these tests in place will help discover when/if further changes break cases like these to aid in writing helpful changesets for impacted customers using these unsupported patterns on accident.
+	 *
+	 * These patterns also [break type safety in .d.ts generation](https://github.com/microsoft/TypeScript/issues/55832):
+	 * this is one of the reasons they are not supported.
+	 * The import-testing package has test coverage for aspect.
```
Comment (CraigMacomber) [inline packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts:770]:
```suggestion
	 * The import-testing package has test coverage for this aspect.
```

File: packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts (hunk 5 above)
Comment (CraigMacomber) [reply packages/dds/tree/src/test/simple-tree/api/schemaFactoryRecursive.spec.ts]: should be clearer now

**Review by jenn-le — APPROVED** [review]
